### PR TITLE
API /config

### PR DESCRIPTION
**Pull request by kylerberry**
- shareable bot, and manager instances
- ConfigProvider for on-the-fly updates
- more test coverage
- connects @ZurabWeb /config route to existing server connection

👍 

run `npm run jact`
navigate to `localhost:PORT/config` you will see the currently used config

screen shot 2018-02-24 at 3 51 08 pm

### CODE REVIEW

**Comment by kylerberry** [comment]
@ZurabWeb I took your config route and added it to the bot's app.js. I also removed `bin/www` because it's basically doing manually what express does auto-magically.

**Comment by ZurabWeb** [comment]
@kylerberry Tests are failing

File: app.js [hunk 1]
```diff
@@ -1,88 +1,92 @@
...
-const FeedService = require('./lib/FeedService')
+// catch 404 and forward to error handler
+app.use(function (req, res, next) {
+	var err = new Error('Not Found');
+	err.status = 404;
+	next(err);
+});
 
-let Spinner = require('cli-spinner').Spinner
-let spinner = new Spinner('working... %s')
```
Comment (ZurabWeb) [inline app.js:14]: @kylerberry Getting rid of the spinner?

File: backtest.js [hunk 2]
```diff
@@ -1,11 +1,10 @@
...
 const app = require('express')()
-const config = require('./lib/config')
-
-const { gdax, DANGER_LIVE_GDAX_DANGER } = require('./lib/gdax')
-
-const Strategy = require('./lib/Strategy')
-const PortfolioManager = require('./lib/PortfolioManager')
+const find = require('lodash/find')
+const config = require('./lib/ConfigProvider')
+const { gdax } = require('./lib/gdax')
```
Comment (ZurabWeb) [inline backtest.js:5]: @kylerberry Probably don't need the curly brackets around `gdax`

File: app.js (hunk 1 above)
Comment (kylerberry) [reply app.js:14]: yea, it never really worked reliably and now that we're moving toward a Web UI view I don't think it's necessary

File: backtest.js (hunk 2 above)
Comment (kylerberry) [reply backtest.js:5]: @ZurabWeb I do. It's ES6 deconstruct syntax. 

`require('./lib/gdax')` returns an object like: `{ gdax: gdaxClient, websocket: websocketClient }`

`{ gdax }` will deconstruct it to just return the gdaxClient

**Comment by kylerberry** [comment]
@ZurabWeb i have to look into why the tests are failing because they pass locally

**Comment by kylerberry** [comment]
@ZurabWeb huzzah! tests are passing

**Pull request merged**